metals/src/main/scala/scala/meta/internal/builds/WorkspaceReloadStatus.scala

```diff
@@ -0,0 +1,33 @@
+package scala.meta.internal.builds
+
+import scala.meta.internal.builds.Digest.Status
+
+/**
+ * Represents the the status of a `workspace/reload` or a `bloopInstall` request.
+ */
+sealed abstract class WorkspaceReloadStatus extends Product with Serializable {
+  import WorkspaceReloadStatus._
+  def name: String =
+    this match {
+      case Duplicate(status) => status.toString
+      case _ => this.toString
+    }
+  def isInstalled: Boolean = this == Installed
+  def isFailed: Boolean = this.isInstanceOf[Failed]
+  def toChecksumStatus: Option[Status] =
+    Option(this).collect {
+      case WorkspaceReloadStatus.Rejected => Status.Rejected
+      case WorkspaceReloadStatus.Installed => Status.Installed
+      case WorkspaceReloadStatus.Cancelled => Status.Cancelled
+      case Failed(_) => Status.Failed
+    }
+}
+object WorkspaceReloadStatus {
```

Comment:
Why not something like `WorkspaceConnectionStatus`. `Reload` seems to suggest that it has only something to do with the `workspace/reload` method, while this is not connected to that? Correct me if I am wrong, the name just feels confusing to me.

Reply:
> `Reload` seems to suggest that it has only something to do with the `workspace/reload` method

Yup, this does actually. Previously this was only used in `BloopInstall` so the wording was really bloop specific. Now we use this for the status of how `BloopInstall` went or the status of a `workspace/reload`. I went with the reload since it was closer to the spec naming. I wouldn't really go towards `WorkspaceConnectionStatus`, since it's not really signifying the connect as much as it is the result of the _install_ or _reload_.

Reply:
`WorkspaceLoadStatus` ? But, it's really just a nitpick, we can leave as is.

Reply:
Haha, no it's a fair point. I ended up landing on `WorkspaceLoadedStatus`.